### PR TITLE
Track invalidations in CI, report if they increase

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by fingolfin** [comment]
@lgoettgens just in case you are bored :joy:, this is another CI test we could potentially apply to all our repositories

File: .github/workflows/Invalidations.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      with:
+        ref: ${{ github.event.repository.default_branch }}
+    - uses: julia-actions/julia-buildpkg@v1
+    - uses: julia-actions/julia-invalidations@v1
+      id: invs_default
+
+    - name: Report invalidation counts
+      run: |
+        echo "Invalidations on default branch: ${{ steps.invs_default.outputs.total }} (${{ steps.invs_default.outputs.deps }} via deps)" >> $GITHUB_STEP_SUMMARY
+        echo "This branch: ${{ steps.invs_pr.outputs.total }} (${{ steps.invs_pr.outputs.deps }} via deps)" >> $GITHUB_STEP_SUMMARY
```
Comment (lgoettgens) [inline .github/workflows/Invalidations.yml:37]: This line does not show up in the logs. See https://github.com/Nemocas/AbstractAlgebra.jl/actions/runs/5141607690/jobs/9254292093?pr=1380#step:9:1

**Comment by lgoettgens** [comment]
> @lgoettgens just in case you are bored joy, this is another CI test we could potentially apply to all our repositories

I would keep it just on AA until we have tested it a little bit (i.e. used it on some PRs). After that, I can surely do that

**Comment by lgoettgens** [comment]
It is possible / easy / sensible to print the invalidations to the log? This could maybe help in investigating why the number increased. Of course only if it is a small order of magnitude of prints :sweat_smile:

File: .github/workflows/Invalidations.yml (hunk 1 above)
Comment (thofma) [reply .github/workflows/Invalidations.yml:37]: I can see it? https://github.com/Nemocas/AbstractAlgebra.jl/actions/runs/5141607690/jobs/9254292093?pr=1380#step:9:2

Or do you mean something else?

Comment (lgoettgens) [reply .github/workflows/Invalidations.yml:37]: I could have sworn that was not there when I last looked at it :sweat_smile:

**Comment by fingolfin** [comment]
@lgoettgens I hope they'll merge https://github.com/julia-actions/julia-invalidations/pull/11 which will make it rather easy to review the invalidations

**Comment by lgoettgens** [comment]
> @lgoettgens just in case you are bored joy, this is another CI test we could potentially apply to all our repositories

Done

**Pull request merged**